docs/contribute/refactoring-filesystem.md

```diff
@@ -0,0 +1,105 @@
+# Refactoring of Saros Filesystem
+
+During the beginning of the development of Saros for IntelliJ (Saros/I), some components of Saros/E were extracted to an undependend component (Saros Core). One of the components is the Saros filesystem: the resources.
+The design of the Saros filesystem is very simliar to the filesystem of Eclipse IDE.
+The Eclipse filesystem can be read here ([here](https://help.eclipse.org/2019-09/topic/org.eclipse.platform.doc.isv/guide/resInt.htm)).
+
+Because the design of the filesystem is similar, so that the implementiation of it for Saros/E was easy.
+With the implementation of Saros/I it turned out, that the design of the Saros filesystem is not suitable for Saros/I.
+It has several reason, for example, that IntelliJ module can be encapsulate in other IntelliJ modules or the module can have several content roots.
+The structure of the IntelliJ filesystem can be read here ([here](https://www.jetbrains.com/help/idea/creating-and-managing-modules.html)) and the comparison to the Eclipse filesystem can be read here ([here](https://www.jetbrains.com/help/idea/migrating-from-eclipse-to-intellij-idea.html))
+
+
+## The new approach
+
+The new approach is to leave the project-based architecture of Saros:
+
+A Saros user should be able to share an any subtree of hin resources with Saros. The root of the subtree must have an absolute path on each Saros user's enviroment. It doesn't matter, where the the subtree is located (ex. On *C*: oder *D*: drive), but what the absolute pathes unites is, that they mean the same reference (IReferencePoint). Every Saros instance should know, where an IReferencePoint object is located in the local file system, so that the resources can be determined in combination of an IReferencePoint and a relative path to the resource.
+
+The IReferencePoint can be look like this:
+
+> public interface IReferencePoint {
+>          /* Nothing here */
+>          }
+
+Every resource knows their reference point, so that a reference point is easy accessable, but not the other way around.
+
+### Current Situation
+The current situation is, that all resources belongs to their projects. 
+For example: all actions of a Saros user, like editing a file, creating a new folder, move files) are called activities. 
+The acitivities contains a global project id (all shared project gets an unique id for all Saros instances in a session) and a relative path to the affected resource outgoing from the project. Every Saros instance knows which id belongs to which project, so it can process the activities and determining the resources easily.
+
+### Design of new filesystem
+Because every Saros instance knows, where the reference point is located in the local filesystem, the Saros' filesystem can be simplified. Saros should not know anymore, if a resource is `IWorkspace` or `IProject`. Saros must distinguish resources between files and folders, so that the design of the filesystem can look like this:
```

Comment:
```suggestion
Because every Saros instance knows, where the reference point is located in the local filesystem, the Saros' filesystem can be simplified. Saros should not know anymore if a resource is `IWorkspace` or `IProject`. Saros must distinguish resources between files and folders so that the design of the filesystem can look like this:
```